### PR TITLE
Add option to pass parameters to the loss function.

### PR DESCRIPTION
**Pull request by PhilipMay**
This PR is by far not finished. In the current state it is just a demo how I would implement it. See #7024

- loss_function_params is a dict that gets passed to the CrossEntropyLoss constructor
- that way you can set class_weights for the `CrossEntropyLoss ` for example
- this is a very important thing to do when working with unbalanced data and can improve your metric by a large amount
- it should be no breaking change since `loss_function_params` is always initialized with `None` which is just the current behavior

Example:
```python
model_name = 'bert-base-german-dbmdz-uncased'

config = AutoConfig.from_pretrained(
    model_name,
    num_labels=3,
)

model = AutoModelForSequenceClassification.from_pretrained(
    model_name,
    config=config,
    loss_function_params={"weight": [0.8, 1.2, 0.97]}
)
```

## ToDo
- [ ] discuss with @LysandreJik @sgugger @nvs-abhilash
- [ ] implement for other models
- [ ] add docstrings
- [ ] write tests
- [ ] maybe implement for TF also

### CODE REVIEW

**Comment by PhilipMay** [comment]
Closed by this answer: https://github.com/huggingface/transformers/issues/7024#issuecomment-691481436
